controllers/storageclient_controller.go

```diff
@@ -78,18 +78,26 @@ type StorageClientReconciler struct {
 
 // SetupWithManager sets up the controller with the Manager.
 func (s *StorageClientReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctx := context.Background()
 	// Index should be registered before cache start.
 	// IndexField is used to filter out the objects that already exists with
 	// status.phase != failed This will help in blocking
 	// the new storageclient creation if there is already with one with same
 	// provider endpoint with status.phase != failed
-	_ = mgr.GetCache().IndexField(context.TODO(), &v1alpha1.StorageClient{}, "spec.storageProviderEndpoint", func(o client.Object) []string {
+	_ = mgr.GetCache().IndexField(ctx, &v1alpha1.StorageClient{}, "spec.storageProviderEndpoint", func(o client.Object) []string {
 		res := []string{}
 		if o.(*v1alpha1.StorageClient).Status.Phase != v1alpha1.StorageClientFailed {
 			res = append(res, o.(*v1alpha1.StorageClient).Spec.StorageProviderEndpoint)
 		}
 		return res
 	})
+
+	if err := mgr.GetCache().IndexField(ctx, &v1alpha1.StorageClassClaim{}, storageClientAnnotationKey, func(obj client.Object) []string {
```

Comment:
@Madhu-1 @leelavg I think we need to make an effort to have a convention for indexers' names. 
Right now we have `spec.xxx` for one field and then `ocs.openshift.io/xxx` for another. 

I think the best way this whole and to give context on the list operation is to use an index prefix (as we are listing on an index not a filed) and then a meaningful index name.


So `spec.storageProviderEndpoint` becomes `index:storageProviderEndpoint` and for the annotation, we can have `index:storageClientAnnotation`

I also think we should use literals and not constants, or a constant like `storageClientAnnotationIndexName`  in the list operation because we want to have it very visible that we are listing on an indexer in the list call site

Reply:
> I also think we should use literals and not constants, or a constant like storageClientAnnotationIndexName in the list operation because we want to have it very visible that we are listing on an indexer in the list call site
- I suppose you mean, if we use constant we want to make it obvious that we are using the field for index or we don't want const at all?